text/3383-recommended-bin-crates.md

```diff
@@ -0,0 +1,162 @@
+
+The developers of `foobar` suggest you may want to install `foobar-cli` instead.
+```
+
+(Notice the last line in the above output, which is enabled by the
+`recommended-bin-crates` field.)
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The `cargo install` command has already parsed the `Cargo.toml` manifest
+file when it prints this error message, so it would simply have to
+additionally check for this new field when printing the error message.
+
+`cargo publish` should assert that the referenced
+crates already exist and have not been yanked.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* It introduces yet another manifest `field`.
+
+* The crates referenced by this field could become abandoned, out-of-date or yanked
+  (although the referencing and referenced crate will presumably often be published
+  by the same person/group, in which case this can be simply avoided).
+
+* Like any other manifest field, updating the field requires you to publish
+  a new version of the crate (a patch version bump suffices).
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The problem addressed by this RFC can be sidestepped by publishing the
+library along with the binary in a single crate. This does however come
+with two disadvantages:
+
+* Cargo currently doesn't support artifact-specific dependencies
+  (although that may change, see [RFC 2887] & [RFC 3374]).
+
+* You would have to bump the library version each time you want to
+  publish a new version of the binary. If you want independently
+  incrementable versions for your library and your binary, you have to
+  publish them in separate crates.
+
+The problem could also be sidestepped by publishing the command in a
+crate with the same name as the command and using a different name for
+the library crate. (While `cargo add` currently does not fail for
+binary-only crates that could very well be addressed without an RFC.)
+A disparity between crate name and library name again comes with its
+own disadvantages, e.g. Rust users often expect libraries published
+on crates.io to be documented at docs.rs/{lib_name}.
+
+So while these two alternatives have their own disadvantages, they are
+also generally not an option for projects that have already been published
+using the described naming convention, since converting a library-only
+crate to a binary-only crate is bound to result in confusion.
+
+The proposed field provides a simple solution to a rather annoying
+problem and could be easily applied to existing projects using this
+common naming convention to solve the described user experience problem.
+
+# Prior art
+[prior-art]: #prior-art
+
+Most other package managers do not have such a problem since their
+install command does not mandate that the package contains binaries.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+* Is `recommended-bin-crates` a good name for the field?
```

Comment:
`recommended-bin-packages`, instead? Casual confusion between crates and packages is ubiquitous, but it ought not to be in the formal structure. And in this case, it may be particularly important to clarify that it is referring to the package **namespace**.

Reply:
Ah right ... very good point, I'll rename it :)